sdk/provisioning/Azure.Provisioning/tests/Expressions/ArrayExpressionTests.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Azure.Provisioning.Expressions;
+using NUnit.Framework;
+
+namespace Azure.Provisioning.Tests.Expressions
+{
+    public class ArrayExpressionTests
+    {
+        [TestCaseSource(nameof(ValidateArrayExpressionTestData))]
+        public string ValidateArrayExpressions(ArrayExpression expression)
+        {
+            return expression.ToString();
+        }
+
+        private static IEnumerable<TestCaseData> ValidateArrayExpressionTestData()
+        {
+            // NOTE: our AST always produces multi-line arrays with indents
+            // empty array
+            yield return new TestCaseData(new ArrayExpression())
+                .Returns($"[]");
+            // array of literals
+            yield return new TestCaseData(new ArrayExpression(new IntLiteralExpression(314), new BoolLiteralExpression(false), new StringLiteralExpression("foo")))
+                .Returns(@"[
+  314
+  false
+  'foo'
+]");
```

Comment:
Style nit - prefer `"""` over `@"` because it's much nicer.  It lets you manage indents better and [it does a MUCH better job of letting you interpolate with things like JSON](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated#interpolated-raw-string-literals).

These two strings are equivalent:
```csharp
        Assert.AreEqual(
            @"[
  314
  false
  'foo'
]",
            """
            [
              314
              false
              'foo'
            ]
            """);
```